File: nativeLib/rary.meteorological/src/ptozsa.f
      Real*4 Function PtoZsa(p)
 
C*  This routine converts a pressure in milibars into a height in a
C*  standard atmosphere in meters.
 
C Author: J Ramer  written in the late 1980's

      Implicit None
 
      Real*4 T0,gamma,p0,p11,z11,c1,c2,p
 
      Data T0,gamma,p0/288.,.0065,1013.2/
      Data c1,c2/5.256,14600./
      Data z11,p11/11000.,226.0971/
 
      Real*4 Flag,Flg
      Common/FlagFlg/Flag,Flg
      Data Flag,Flg/1e37,1.0E10/
 
      If (p.gt.Flg .or. p.lt.1.0) Then
          PtoZsa=Flag
        Else If (p.gt.p11) Then
          PtoZsa=((T0-T0*(p/p0)**(1/c1))/gamma)
        Else
          PtoZsa=(c2*alog10(p11/p)+z11)
       End If
 
      Return
      End
 

File: edexOsgi/com.raytheon.uf.common.wxmath/src/com/raytheon/uf/common/wxmath/PToZsa.java
/**
 * This software was developed and / or modified by Raytheon Company,
 * pursuant to Contract DG133W-05-CQ-1067 with the US Government.
 * 
 * U.S. EXPORT CONTROLLED TECHNICAL DATA
 * This software product contains export-restricted data whose
 * export/transfer/disclosure is restricted by U.S. law. Dissemination
 * to non-U.S. persons whether in the United States or abroad requires
 * an export license or other authorization.
 * 
 * Contractor Name:        Raytheon Company
 * Contractor Address:     6825 Pine Street, Suite 340
 *                         Mail Stop B8
 *                         Omaha, NE 68106
 *                         402.291.0100
 * 
 * See the AWIPS II Master Rights File ("Master Rights File.pdf") for
 * further licensing information.
 **/
package com.raytheon.uf.common.wxmath;

/**
 * Converts a pressure in milibars into a height in a standard atmosphere in
 * meters.
 * 
 * <pre>
 * 
 * SOFTWARE HISTORY
 * 
 * Date         Ticket#    Engineer    Description
 * ------------ ---------- ----------- --------------------------
 * Aug 13, 2013  #2262     dgilling     Ported from ptozsa.f.
 * 
 * </pre>
 * 
 * @author dgilling
 * @version 1.0
 */

public class PToZsa {

    private static final double Flg = 1e10;

    // Never allow this class to be directly instantiated
    private PToZsa() {
        throw new AssertionError();
    }

    /**
     * This routine converts a pressure in milibars into a height in a standard
     * atmosphere in meters.
     * 
     * @param pressure
     *            Pressure (in mb)
     * @return Height (in m)
     */
    public static float ptozsa(float pressure) {
        float PtoZsa;
        if (pressure > Flg || pressure < 1.0) {
            PtoZsa = Float.NaN;
        } else if (pressure > Constants.p11) {
            PtoZsa = (float) ((Constants.T0 - (Constants.T0 * Math.pow(
                    (pressure / Constants.p0), (1 / Constants.HGT_PRES_c1)))) / Constants.gamma);
        } else {
            PtoZsa = (float) (Constants.HGT_PRES_c2
                    * Math.log10(Constants.p11 / pressure) + Constants.z11);
        }

        return PtoZsa;
    }
}
